src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1869,24 +1891,34 @@ bool isRefEscape
                 isRefEscape,
                 ignoreArglistRefKinds: true, // https://github.com/dotnet/roslyn/issues/63325: for compatibility with C#10 implementation.
                 argsAndParamsAll);
+
+            var returnsWritableRefToRefStruct = ReturnsWritableRefToRefStruct(symbol);
             foreach (var argAndParam in argsAndParamsAll)
             {
-                var argument = argAndParam.Argument;
-                bool valid = argAndParam.IsRefEscape ?
-                    CheckRefEscape(argument.Syntax, argument, escapeFrom, escapeTo, false, diagnostics) :
-                    CheckValEscape(argument.Syntax, argument, escapeFrom, escapeTo, false, diagnostics);
-
-                if (!valid)
+                // SPEC:
+                // If `M()` does return ref-to-ref-struct, the *safe-to-escape* is the same as the *safe-to-escape* of all arguments which are ref-to-ref-struct.
```

Comment:
>. [](http://example.com/codeflow?start=160&length=1)

Consider adding "It is an error if there are multiple arguments with different *safe-to-escape* because of *method arguments must match*."